README.md

```diff
@@ -112,84 +125,141 @@ The content of your document is retrieved.
 *   [decodeDocument](#decodedocument)
     *   [Parameters](#parameters-2)
 *   [IEncryptionResults](#iencryptionresults)
-*   [encryptString](#encryptstring)
-    *   [Parameters](#parameters-3)
-*   [decryptString](#decryptstring)
-    *   [Parameters](#parameters-4)
+    *   [encryptString](#encryptstring)
+        *   [Parameters](#parameters-3)
+    *   [decryptString](#decryptstring)
+        *   [Parameters](#parameters-4)
 
 ### ENCRYPTION\_PARAMETERS
 
-Default options from responses
-[here](https://crypto.stackexchange.com/questions/26783/ciphertext-and-tag-size-and-iv-transmission-with-aes-in-gcm-mode/26787)
+To learn more about the default options from responses, see this [StackExchange post](https://crypto.stackexchange.com/questions/26783/ciphertext-and-tag-size-and-iv-transmission-with-aes-in-gcm-mode/26787).
 
 #### algorithm
 
+The algorithm is Galois/Counter Mode (GCM). It aims for authenticated encryption with associated data; and its specialization GMAC, for generating a
+message authentication code (MAC) on data that is not encrypted.
+
+GCM and GMAC are modes of operation for an underlying approved symmetric key block cipher. 
+
 #### keyLength
 
+The key size should be at least 128 bits. See this [NIST publication](http://csrc.nist.gov/publications/nistpubs/800-38D/SP-800-38D.pdf) to learn more.
+
 #### ivLength
 
+In Galois/Counter Mode (GCM), a 12 byte IV is strongly suggested as other IV lengths will require additional calculations. 
+
+In principle, any IV size can be used as long as the IV doesn't repeat. However, [NIST](http://csrc.nist.gov/publications/nistpubs/800-38D/SP-800-38D.pdf) suggests only an IV size of 12 bytes needs to be supported in the implementations.
+
 #### tagLength
 
+Use a tag size of 64 bits at the very minimum, but it is preferable to use a tag size of the full 128 bits.
+
 #### version
 
+<!--Flag: Through a search in the NIST publication and the post on stackexchange, version was not found. We need to tell the users which version it is talking about.-->
```

Comment:
You can refer to my previous comment for the default options

Reply:
added `"OPEN-ATTESTATION-TYPE-1"`